### PR TITLE
Mounted config files docker compose and sped up image build times

### LINKED ISSUE #28: Docker compose mounted directories creates rebuild issues
Changes to the `src` can't be rebuilt once it's been deployed, since the user doesn't have permissions on the `docker-db-data` directory which is sent to the docker build context.  This causes `docker-compose up --build` to complain of a docker socket issue when it's actually masking the permissions error (evident by running `docker build .` in the project root).

I've fixed this locally by adding the folder to the `.dockerignore` since it shouldn't be sent to the image anyway.

### LINKED ISSUE #29: Mount in config.yaml to maplestory service
Currently the config is built with the jar, but mounting it in would make it easier to redeploy and test changes, since it shouldn't need a full rebuild to be applied.

### PR DESCRIPTION
**Pull request by wejrox**
Fixes #29 
Fixes #28

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,20 +1,42 @@
...
+WORKDIR /opt/cosmic
+# Any changes to the pom will affect the entire build, so it should be copied first.
+COPY pom.xml ./pom.xml
+# Grab all the dependencies listed in the pom early, since it prevents changes to source code from requiring a complete re-download.
+# Skip compiling tests since we don't want all the dependecies to be downloaded.
+RUN mvn -f ./pom.xml clean dependency:go-offline -Dmaven.test.skip -T 1C
+# Source code changes may not change dependencies, so it can go last.
+# Skip compiling tests since we don't want all the dependecies to be downloaded for plugins.
+COPY src ./src
+RUN mvn -f ./pom.xml clean package -Dmaven.test.skip -T 1C
```
Comment (P0nk) [inline Dockerfile:19]: Do you really need to specify `-T 1C`? I had to look it up, and it's to set the thread count to 1, which seems obscure to me. The simpler the better.

**Review by P0nk — CHANGES_REQUESTED** [review]
Thank you for the PR. Glad you could contribute, since you seem more knowledgeable about Docker than me.
I only have a minor issue with ´-T 1C´ in the maven commands.

File: Dockerfile (hunk 1 above)
Comment (wejrox) [reply Dockerfile:19]: It's not required, but the option tells maven to run multithreaded with one thread per core that is available, which speeds up module compilation and package time.

Comment (P0nk) [reply Dockerfile:19]: That's reassuring; PR approved.

**Review by P0nk — APPROVED** [review]

**Pull request merged**